test/integration/connect/envoy/case-cross-peers/primary/verify.bats

```diff
@@ -55,3 +55,7 @@ load helpers
 @test "s1 upstream made 1 connection to s2" {
   assert_envoy_metric_at_least 127.0.0.1:19000 "cluster.s2.default.primary-to-alpha.external.*cx_total" 1
 }
+
+@test "s1 upstream made 1 connection to s2 through the primary mesh gateway" {
+  assert_envoy_metric_at_least 127.0.0.1:19001 "cluster.s2.default.default.alpha-to-primary.external.*cx_total" 1
```

Comment:
This seems unusual, why is `alpha-to-primary` here? That's the name that `alpha` uses to refer to `primary`, so why would the local primary gateway be aware of that?

Reply:
Yeah, I was confused by this but was hoping there was some logic behind it. This matches [the SNI created by the sidecar proxy](https://github.com/hashicorp/consul/blob/main/agent/xds/clusters.go#L919) 🤔

Reply:
Ah I see what's going on. That's how the remote side can know that this traffic is for an exported service as opposed to the usual mesh gateway route for `s2`. See this comment: https://github.com/hashicorp/consul/blob/main/agent/xds/clusters.go#L1284-L1298

I'm a bit worried about whether there could be collisions on the dialing side though.

It encodes: `(service, namespace, partition, peerName, trustDomain)` so I think there could be a collision if there is a separate peering with all the same data but different DC.